src/Features/Blockcore.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -154,7 +154,7 @@ public void ReindexChain(IConsensusManager consensusManager, CancellationToken n
 
             headers.Reverse();
 
-            BlockRepository blockRepository = (BlockRepository)this.blockRepository;
+            RocksdbBlockRepository blockRepository = (RocksdbBlockRepository)this.blockRepository;
```

Comment:
This will crash, won't it?

Reply:
ahh yes good spot! why the tests didn't pick this...

Reply:
ah its just making he method as part of the interface (well spotted @sondreb )

Reply:
to prevent this kind of problems, I'd suggest to move Rocksdb implementation in its own feature and just use interfaces

Reply:
to be clear, an example of project can be `Blockcore.Features.Blockstore.Rocksdb`

Reply:
You probably mean its own library not its own feature (feature means we need to impl the blockstore logic again)